builder/job.go

```diff
@@ -46,6 +46,16 @@ var validCommitCommands = map[string]bool{
 	"workdir":    true,
 }
 
+// BuiltinAllowedBuildArgs is list of built-in allowed build args
+var BuiltinAllowedBuildArgs = map[string]bool{
+	"HTTP_PROXY":  true,
+	"HTTPS_PROXY": true,
+	"http_proxy":  true,
+	"https_proxy": true,
+	"FTP_PROXY":   true,
+	"NO_PROXY":    true,
```

Comment:
actually I didn't know. I added the lower case http_proxy and https_proxy as that's what I use in my proxy environment.

But a quick google search [shows](https://wiki.archlinux.org/index.php/Proxy_settings) that the lower cases version of  `FTP_PROXY` and `NO_PROXY` are valid as well. I will add them to the list.